### PR TITLE
Crop blending 

### PR DESCRIPTION
**Pull request by MartinHasal**
Repaired:
- moved check for number of iterations in warm-up option
- reaired types for arguments, added Range(0,1) check
- added .git saving to .jpg. OpenCV nor PIL saves .gif properly
- repaired error with ds_test

Added functions:
- clean image, which takes a threshold (percentual size of image) and remove all smaller unconnected object
- plotListOfImages as input takes prediction (output from inference) and visualizes output, allows blending and clean image. Basically it is preparation for model signature
- predictListOfFiles - predicts from list of paths

Changed:
plotColorizedVessels - collorized vessels handle blending
README.md - written readme, images will be added later

### CODE REVIEW

File: utils/plots.py [hunk 1]
```diff
@@ -7,6 +7,9 @@
 import seaborn as sns
 from PIL import Image
 
+#from funcs.inference import predictListOfFiles
```
Comment (mpecha) [inline utils/plots.py]: Remove unused comments

File: utils/plots.py [hunk 2]
```diff
@@ -193,3 +205,109 @@ def update(val):
...
+            img3 = item['mask_label'] * 255
+            img3 = img3.astype(np.uint8)
+            plt.sca(axs[2]) 
+            plt.imshow(clean_image(img3, percentage=clean_threshold)); 
+            plt.title(f'Mask - probability.\n Treshold {clean_threshold}')
+        else:        
+            plt.sca(axs[2]); 
+            plt.imshow(item['mask_label']); plt.title('Mask - label')
+        plt.show()
+
```
Comment (mpecha) [inline utils/plots.py:311]: remove empty line

File: funcs/inference.py [hunk 3]
```diff
@@ -107,4 +109,85 @@ def predictDataset(ds, nsamples_to_plot: int, nn_model) -> None:
     plt.show()
 
     #
-    print('auc roc = {0:.4f}'.format(auc_roc.auc))
\ No newline at end of file
+    print('auc roc = {0:.4f}'.format(auc_roc.auc))
+    
+
+"""
```
Comment (mpecha) [inline funcs/inference.py:115]: Would not be this function uncomment?

File: utils/smooth_blender_predicitions.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+    wind_inner = 1 - (abs(2*(scipy.signal.triang(window_size) - 1)) ** power)/2
+    wind_inner[:intersection] = 0
+    wind_inner[-intersection:] = 0
+
+    wind = wind_inner + wind_outer
+    wind = wind / np.average(wind)
+    return wind
+
+
+cached_2d_windows = dict()
```
Comment (mpecha) [inline utils/smooth_blender_predicitions.py:40]: I think, it is better not using global variables

File: utils/smooth_blender_predicitions.py [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+    wind_outer[intersection:-intersection] = 0
+
+    wind_inner = 1 - (abs(2*(scipy.signal.triang(window_size) - 1)) ** power)/2
+    wind_inner[:intersection] = 0
+    wind_inner[-intersection:] = 0
+
+    wind = wind_inner + wind_outer
+    wind = wind / np.average(wind)
+    return wind
+
```
Comment (mpecha) [inline utils/smooth_blender_predicitions.py:38]: Between functions should be 1 empty line.

**Review by mpecha — CHANGES_REQUESTED** [review]
It looks great. However, there are some minor issues - code not satisfying PEP.

File: utils/plots.py (hunk 1 above)
Comment (MartinHasal) [reply utils/plots.py]: Thanks, my mistake.

File: funcs/inference.py (hunk 3 above)
Comment (MartinHasal) [reply funcs/inference.py:115]: No, it is a commented/unused signature function. I will use it later, maybe.

File: utils/smooth_blender_predicitions.py (hunk 5 above)
Comment (MartinHasal) [reply utils/smooth_blender_predicitions.py:38]: Please, among which functions should be one empty line? I checked all definitions of functions, and two lines separate all.

File: utils/smooth_blender_predicitions.py (hunk 4 above)
Comment (MartinHasal) [reply utils/smooth_blender_predicitions.py:40]: Yes, you are right. Especially here, it seems not to be necessary. I noted it. I will resolve this directly in the main after merging and after testing.

**Review by MartinHasal — COMMENTED** [review]
Branch accepted. One comment needs response. One note will be solved later.

File: utils/smooth_blender_predicitions.py (hunk 5 above)
Comment (MartinHasal) [reply utils/smooth_blender_predicitions.py:38]: Solved by talk.

**Review by mpecha — APPROVED** [review]

**Comment by MartinHasal** [comment]
Checked.

**Pull request merged**